examples/request_state_local_cache/src/main.rs

```diff
@@ -0,0 +1,59 @@
+#![feature(plugin, decl_macro)]
+#![plugin(rocket_codegen)]
+extern crate rocket;
+
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use rocket::request::{self, Request, FromRequest, State};
+use rocket::outcome::Outcome::*;
+
+#[cfg(test)] mod tests;
+
+#[derive(Default)]
+struct Atomics {
+    uncached: AtomicUsize,
+    cached: AtomicUsize,
+}
+
+struct Guard1;
+struct Guard2;
+
+impl<'a, 'r> FromRequest<'a, 'r> for Guard1 {
+    type Error = ();
+
+    fn from_request(req: &'a Request<'r>) -> request::Outcome<Self, ()> {
+        req.local_cache(|| {
```

Comment:
Please make the change in the previous comment.

Reply:
I see I commited my experiments.